components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientByExternalIdResourceImplTest.java

```diff
@@ -0,0 +1,345 @@
+        when(this.repository.getPatientByExternalId(this.eid)).thenReturn(this.patient);
+        when(this.access.hasAccess(Right.VIEW, null, null)).thenReturn(true);
+
+        JSONObject jsonObject = new JSONObject();
+        when(this.patient.toJSON()).thenReturn(jsonObject);
+        when(this.uriInfo.getBaseUriBuilder()).thenReturn(this.uriBuilder);
+        when(this.uriBuilder.path(PatientResource.class)).
+                thenReturn(uriBuilder);
+        when(this.uriBuilder.build(this.patient.getId())).thenReturn(this.uri);
+
+        Response response = this.component.getPatient(this.eid);
+        verify(this.logger).debug("Retrieving patient record with external ID [{}] via REST", this.eid);
+
+        JSONObject links = new JSONObject().accumulate("rel", Relations.SELF).accumulate("href", "uri");
+        JSONObject json = new JSONObject().accumulate("links", links);
+
+        assertEquals(json, response.getEntity());
+        assertEquals(Response.Status.OK.getStatusCode(), response.getStatus());
+    }
+
+    @Test
+    public void getPatientNotFoundChecksForMultipleRecords() throws ComponentLookupException, QueryException, XWikiRestException
+    {
+        Query query = mock(DefaultQuery.class);
+        when(this.repository.getPatientByExternalId(this.eid)).thenReturn(null);
+        when(this.qm.createQuery(Matchers.anyString(), Matchers.anyString())).thenReturn(query);
+        when(query.execute()).thenReturn(new ArrayList<Object>());
+
+        Response response = this.component.getPatient(this.eid);
+        verify(this.logger).debug("No patient record with external ID [{}] exists yet", this.eid);
+        assertEquals(Response.Status.NOT_FOUND.getStatusCode(), response.getStatus());
+    }
+
+    @Test
+    public void updatePatientNotFoundChecksForMultipleRecords() throws ComponentLookupException, QueryException, XWikiRestException
+    {
+        Query query = mock(DefaultQuery.class);
+        when(this.repository.getPatientByExternalId(this.eid)).thenReturn(null);
+        when(this.qm.createQuery(Matchers.anyString(), Matchers.anyString())).thenReturn(query);
+        when(query.execute()).thenReturn(new ArrayList<Object>());
+
+        Response response = this.component.updatePatient("json", this.eid);
+        verify(this.logger).debug("No patient record with external ID [{}] exists yet", this.eid);
+        assertEquals(Response.Status.NOT_FOUND.getStatusCode(), response.getStatus());
+    }
+
+    @Test(expected=WebApplicationException.class)
+    public void updatePatientNoAccessReturnsForbiddenCode() throws ComponentLookupException, XWikiRestException
+    {
+        when(this.repository.getPatientByExternalId(this.eid)).thenReturn(this.patient);
+        when(this.access.hasAccess(Right.EDIT, null, null)).thenReturn(false);
+
+        Response response = this.component.updatePatient("json", this.eid);
+        verify(this.logger).debug("Updating patient record with external ID [{}] via REST", this.eid);
```

Comment:
@sdumitriu is right, `verify(this.logger)` will not be checked as the expected exception is returned on the line above. Instead you can put it in a `try-catch` block.